Userland/Libraries/LibMedia/Video/VP9/Decoder.cpp

```diff
@@ -146,36 +156,32 @@ DecoderErrorOr<void> Decoder::create_video_frame(FrameContext const& frame_conte
     //        sizes, as the spec seems to prefer that the halved sizes be ceiled.
     u32 decoded_y_width = frame_context.decoded_size(false).width();
     auto decoded_uv_width = frame_context.decoded_size(true).width();
-    Gfx::Size<u32> output_y_size = frame_context.size();
-    auto subsampling_x = frame_context.color_config.subsampling_x;
-    auto subsampling_y = frame_context.color_config.subsampling_y;
-    Gfx::Size<u32> output_uv_size = {
-        y_size_to_uv_size(subsampling_x, output_y_size.width()),
-        y_size_to_uv_size(subsampling_y, output_y_size.height()),
-    };
-    Array<FixedArray<u16>, 3> output_buffers = {
-        DECODER_TRY_ALLOC(FixedArray<u16>::create(output_y_size.width() * output_y_size.height())),
-        DECODER_TRY_ALLOC(FixedArray<u16>::create(output_uv_size.width() * output_uv_size.height())),
-        DECODER_TRY_ALLOC(FixedArray<u16>::create(output_uv_size.width() * output_uv_size.height())),
+
+    Subsampling subsampling { frame_context.color_config.subsampling_x, frame_context.color_config.subsampling_y };
+    auto output_y_size = frame_context.size().to_type<size_t>();
+    auto output_uv_size = subsampling.subsampled_size(output_y_size);
+
+    Array<FixedArray<u8>, 3> output_buffers = {
+        DECODER_TRY_ALLOC(FixedArray<u8>::create(output_y_size.width() * output_y_size.height() * sizeof(T))),
+        DECODER_TRY_ALLOC(FixedArray<u8>::create(output_uv_size.width() * output_uv_size.height() * sizeof(T))),
+        DECODER_TRY_ALLOC(FixedArray<u8>::create(output_uv_size.width() * output_uv_size.height() * sizeof(T))),
     };
     for (u8 plane = 0; plane < 3; plane++) {
-        auto& buffer = output_buffers[plane];
+        auto* buffer = reinterpret_cast<T*>(output_buffers[plane].data());
```

Comment:
any alignment concerns with this data here?

Reply:
Hopefully fixed by using aligned_alloc based on the bit depth.